caddyhttp/proxy/upstream.go

```diff
@@ -416,9 +423,21 @@ func (u *staticUpstream) healthCheck() {
 			}
 
 			if r, err := u.HealthCheck.Client.Do(req); err == nil {
-				io.Copy(ioutil.Discard, r.Body)
-				r.Body.Close()
-				unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
+				if u.HealthCheck.ContentString != "" {
+					b, err := ioutil.ReadAll(r.Body)
```

Comment:
@reds It was in general, yup. No worries, I posted all the relevant links here.

I don't think this ReadAll() is a showstopper. We can merge without changing it. But I did want to at least document this for future reference. Maybe put a TODO and a link to this discussion? Mention in a comment that we need not make any changes to this until it has been deemed necessary for some reason.